### PR TITLE
WIP: Add selective serialization feature to /write_state.

### PR DESCRIPTION
**Pull request by MichaelGrupp**
Follow googlecartographer/cartographer/#949

### CODE REVIEW

**Comment by MichaelGrupp** [comment]
The re-mapping use-case can be tested e.g. with the localization demo of https://github.com/magazino/cartographer_magazino

You need to set `localization = false,` in `toru_localization.lua` to disable submap trimming.

File: cartographer_ros/cartographer_ros/map_builder_bridge.h [hunk 1]
```diff
@@ -71,7 +71,8 @@ class MapBuilderBridge {
       const TrajectoryOptions& trajectory_options);
   void FinishTrajectory(int trajectory_id);
   void RunFinalOptimization();
-  bool SerializeState(const std::string& filename);
+  bool SerializeState(const std::string& filename,
+                      const std::set<int> trajectory_ids = {});
```
Comment (ojura) [inline cartographer_ros/cartographer_ros/map_builder_bridge.h]: AFAIK Google code style does not permit default values.

File: cartographer_ros_msgs/srv/WriteState.srv [hunk 2]
```diff
@@ -13,5 +13,6 @@
 # limitations under the License.
 
 string filename
+uint32[] trajectory_ids
```
Comment (ojura) [inline cartographer_ros_msgs/srv/WriteState.srv]: I do not like having to specify trajectory ids by default. Perhaps change this to which trajectory IDs to exclude, so the default is "serialize everything"?

**Comment by MichaelGrupp** [comment]
Closing in favor of WIP on new trimmers.

**Pull request closed without merging**